src/Interpreter/Executor.cpp

```diff
@@ -2,16 +2,38 @@
 #include "caffeine/Interpreter/Interpreter.h"
 #include "caffeine/Interpreter/Store.h"
 
+#include <iostream>
+#include <thread>
+
 namespace caffeine {
 
+void run_worker(Executor* exec, FailureLogger* logger,
+                ExecutionContextStore* store) {
+  while (true) {
+    auto ctx = store->next_context();
+
+    if (!ctx.has_value()) {
+      return;
+    }
```

Comment:
Nit: You can simplify this to
```suggestion
while (auto ctx = store->next_context()) {
```